Source/Noesis.Javascript/JavascriptExternal.cpp

```diff
@@ -52,14 +52,37 @@ JavascriptExternal::JavascriptExternal(System::Object^ iObject)
 {
 	mObjectHandle = System::Runtime::InteropServices::GCHandle::Alloc(iObject);
 	mOptions = SetParameterOptions::None;
-	mMethods = gcnew System::Collections::Generic::Dictionary<System::String ^, WrappedMethod>();
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 JavascriptExternal::~JavascriptExternal()
 {
-	mObjectHandle.Free();
+    mObjectHandle.Free();
+    if (mPersistent.IsEmpty())
+        return;
+    mPersistent.ClearWeak<void>();
+    mPersistent.Reset();

```

Comment:
So someone who naively adds something else to this destructor does not get their code missed.
```suggestion
    if (!mPersistent.IsEmpty()) {
        mPersistent.ClearWeak<void>();
        mPersistent.Reset();
    }

```